cloudslang-cli/src/main/java/io/cloudslang/lang/cli/utils/CompilerHelperImpl.java

```diff
@@ -141,16 +131,58 @@ public String convert(File from) {
                     }
                 }
                 if (emptyContent){
-                    throw new RuntimeException("Inputs / System properties file: " + inputFile + " is empty or does not contain valid YAML content.");
+                    throw new RuntimeException("Inputs file: " + inputFile + " is empty or does not contain valid YAML content.");
                 }
 			} catch(IOException ex) {
-				logger.error("Error loading file: " + inputFile, ex);
+				logger.error("Error loading file: " + inputFile + ". Nested exception is: " + ex.getMessage(), ex);
 				throw new RuntimeException(ex);
 			}
 		}
         return result;
     }
 
+    private Set<SystemProperty> loadPropertiesFromFiles(List<String> files, String[] extensions, String directory) {
+        if(CollectionUtils.isEmpty(files)) {
+            files = loadDefaultFiles(files, extensions, directory, true);
+        }
+        if(CollectionUtils.isEmpty(files)) return new HashSet<>();
+        Set<SystemProperty> result = new HashSet<>();
+        for(String propFile : files) {
+            try {
+                SlangSource source = SlangSource.fromFile(new File(propFile));
+                if (!isExecutable(source)) {
+                    logger.info("Loading file: " + propFile);
+                    Set<SystemProperty> propsFromFile = slang.loadSystemProperties(source);
+                    result.addAll(propsFromFile);
+                } else {
+                    logger.info("Ignoring file: " + source.getName() + " (not a valid properties file)." +
+                            "\n\tTop level keys(" + SlangTextualKeys.FLOW_TYPE + ", " + SlangTextualKeys.OPERATION_TYPE + ") are found.");
+                }
+            } catch(Throwable ex) {
+                String errorMessage = "Error loading file: " + propFile + " nested exception is " + ex.getMessage();
+                logger.error(errorMessage, ex);
+                throw new RuntimeException(errorMessage, ex);
+            }
+        }
+        return result;
+    }
+
+    private List<String> loadDefaultFiles(List<String> files, String[] extensions, String directory, boolean recursive) {
```

Comment:
I think the method should return the File objects instead of returning only the paths and then creating the File again in the calling methods